### PR TITLE
Add tag command

### LINKED ISSUE #175: Keep track of dataset tag
A database should remember the tag it was last assigned to. A user should also be able to set a new tag. We should probably store this separately from `dbase.sqlite` (maybe a `.ddb/tag` text file or `.ddb/tag.json`).

`ddb tag`

Should print the current tag.

`ddb tag <new tag>`

Should assign a new tag.

E.g.

`ddb tag <org>/<dataset>`
`ddb tag <server>/<org>/<dataset>`

If unspecified `<server>` is the default registry.

### PR DESCRIPTION
**Pull request by HeDo88TH**
Closes #175

### CODE REVIEW

File: src/cmd/tag.cpp [hunk 1]
```diff
@@ -15,8 +16,8 @@ void Tag::setOptions(cxxopts::Options &opts) {
     .positional_help("[args]")
     .custom_help("tag [tag]")
     .add_options()
-    ("t,tag", "New tag", cxxopts::value<std::string>()->default_value(""));
-
+    ("t,tag", "New tag", cxxopts::value<std::string>()->default_value(""))
+    ("r,registry", "Registry", cxxopts::value<std::string>()->default_value(""));
```
Comment (pierotofy) [inline src/cmd/tag.cpp:22]: You shouldn't need two fields; a tag can include a registry URL (e.g. testhub1.dronedb.app/admin/dataset).

Comment (HeDo88TH) [reply src/cmd/tag.cpp:22]: I just fixed it. We still need the registry parameter when getting the tag

**Comment by HeDo88TH** [comment]
Command specification

To set tag:
`ddb tag [server/]organization/dataset`

To get tag
`ddb tag [(-r|--registry)=registryurl]`

**Comment by pierotofy** [comment]
A few things:

1. `ddb tag` on a dataset that has no tag returns:

```
# ddb tag
[https://testhub.dronedb.app]
```

?

2. Typing `ddb` should return the list of commands with their descriptions: the part for `ddb tag` is a bit messy?

```
...
Show: ddb tag.
Set: ddb tag <org>/<dataset>
or: ddb tag <server>/<org>/<dataset>
```

Limit the description to one line.

3. Check new lines in the std::cout,when I type:

```
ddb tag pierotofy/test
```

I get:

```
pierotofy/test[piero:/data/test] [uav4geo] $
```

Either don't output anything or output the tag with a new line?

4. I don't understand the need for `-r`. Can you explain the logic here?

5. Can we output the tag always in the same format (e.g. `registry/org/dataset` instead of `[registry] org/dataset`?

**Comment by pierotofy** [comment]
I think I understand the logic of `-r`, but it wasn't needed ? A tag is always unique for a dataset (a la docker). There was no mention of it in #175 either.

**Comment by HeDo88TH** [comment]
I fixed all the things. 
The -r parameter is needed if anyone needs to check the tag associated with a specific registry without setting it

**Comment by HeDo88TH** [comment]
Now command is much simple:

`ddb tag [server/organization/tag]`

without specifying tag it shows the current tag

**Comment by pierotofy** [comment]
> 1. ddb tag on a dataset that has no tag returns:

This is still a thing.

> 2. Typing ddb should return the list of commands with their descriptions: the part for ddb tag is a bit messy?

This is still a thing.

> 3. Check new lines in the std::cout,when I type:

This is still a thing.

> 5. Can we output the tag always in the same format (e.g. registry/org/dataset instead of [registry] org/dataset?

This is still a thing.

Can we please make sure to go through all points when amending pull requests? The point of a review is that I don't have to re-test the same things over and over..

**Comment by pierotofy** [comment]
6. Calling `ddb tag` from a DroneDB subdirectory does not return the proper tag.
7. Calling `ddb tag pierotofy/test` from a DroneDB subdirectory returns:

```
terminate called after throwing an instance of 'nlohmann::detail::parse_error'
  what():  [json.exception.parse_error.101] parse error at line 1, column 1: syntax error while parsing value - unexpected end of input; expected '[', '{', or a literal
Aborted (core dumped)
```

**Comment by HeDo88TH** [comment]
> > 1. ddb tag on a dataset that has no tag returns:
> 
> This is still a thing.
> 
> > 1. Typing ddb should return the list of commands with their descriptions: the part for ddb tag is a bit messy?
> 
> This is still a thing.
> 
> > 1. Check new lines in the std::cout,when I type:
> 
> This is still a thing.
> 
> > 1. Can we output the tag always in the same format (e.g. registry/org/dataset instead of [registry] org/dataset?
> 
> This is still a thing.
> 
> Please make sure to go through all points when amending pull requests!

I already checked:

```
C:\Users\ldile\Pictures\Palazzo>ddb tag
No tag set
```

```
C:\Users\ldile\Pictures\Palazzo>ddb
DroneDB v0.9.8 (git commit 21eda95) - Effortless aerial data management and sharing
Usage:
        ddb <command> [args] [PATHS]

Commands:
        add - Add files and directories to an index.
        build - Initialize and build an index. Shorthand for running an init,add,commit command sequence.
        chattr - Manage database attributes
        clone - Clone a repository into a new directory
        delta - Generate delta between two ddb databases
        geoproj - Project images to georeferenced rasters
        info - Retrieve information about files and directories
        init - Initialize an index. If a directory is not specified, initializes the index in the current directory
        list - List indexed files and directories
        login - Authenticate with a registry.
        logout - Logout from all registries. To logout from a single registry, use the --server option.
        password - Manage database passwords
        remove - Remove files and directories from an index. The filesystem is left unchanged (actual files and directories will not be removed)
        setexif - Modify EXIF values in files.
        share - Share files and folders to a registry
        status - Show files and directories index status compared to the filesystem
        sync - Sync files and directories in the index with changes from the filesystem
        system - Manage ddb
        **tag - Gets or sets the dataset tag.**
        thumbs - Generate thumbnails for images and rasters
        tile - Generate tiles for GeoTIFFs and GeoImages

        -h, --help              Print help
        --version               Print version

For detailed command help use: ddb <command> --help
See https://docs.dronedb.app for more information.
```

```
C:\Users\ldile\Pictures\Palazzo>ddb tag pierotofy/test
Tag set: pierotofy/test
```

```
C:\Users\ldile\Pictures\Palazzo>ddb tag
https://testhub.dronedb.app/pierotofy/test
```

Are you sure you pulled it first? 🤔

**Comment by pierotofy** [comment]
Oh aha. Ups! I did pull, but QT Creator didn't rebuild. Apologies! :pray: 

Only 6. and 7. still apply.

**Comment by HeDo88TH** [comment]
> Oh aha. Ups! I did pull, but QT Creator didn't rebuild. Apologies! 🙏
> 
> Only 6. and 7. still apply.

Fixed in https://github.com/DroneDB/DroneDB/pull/192/commits/b542341e5a56906482cf9b8baab2d2b7502344a5

**Comment by pierotofy** [comment]
Looking good! :tada: 

Thanks @HeDo88TH !

**Pull request merged**